### ISSUE TITLE
NonceVerification overzealous ?

### ISSUE THREAD

**Issue by jrfnl** [WordPress-Coding-Standards/WordPress-Coding-Standards#737]
The following snippet of code triggers the `WordPress.CSRF.NonceVerification` sniff.
```php
if ( ! isset( $_POST['field'] ) ) {
	// Do something (not using the post variable.
}
```
As this is checking whether a variable is **not** set, I believe that the error is thrown erroneously.

Opinions ?

**Comment by jrfnl** [comment]
Related #572

**Comment by westonruter** [comment]
Yeah, I agree. We should only start flagging accessing input vars once the value is being used. An `isset()` or `empty()` call shouldn't qualify as “using”.

**Comment by JDGrimes** [comment]
In this case, because you are checking that the value is not set, the sniff may be a bit overzealous. However, I wouldn't favor making it ignore all `isset()` and `empty()` checks. Even if no submitted values are being used, the fact that user input is being checked at all indicates that a user action is being reacted to, and thus CSRF protection is likely still needed.
